### ISSUE TITLE
Label leakage

### ISSUE THREAD

**Issue by danielwalke** [zuozhaorui/SWnet#2]
If I’m interpreting the code correctly, it seems the same test loader is being used for both early stopping and final evaluation. Ideally, there should be separate validation and test sets for these purposes. Could you clarify if this is the case?
Have you also used the same test loader for your hyperparameter tuning (i.e., learning rate, number of layers, kernel and stride sizes)
